utils/parsing.py

```diff
@@ -129,14 +129,27 @@ def parse_brat_file(txt_file: Path, annotation_file_suffixes: List[str] = None)
             fields = line.split("\t")
 
             ann["id"] = fields[0]
-            ann["text"] = [fields[2]]
             ann["type"] = fields[1].split()[0]
             ann["offsets"] = []
             span_str = remove_prefix(fields[1], (ann["type"] + " "))
+            text = fields[2]
             for span in span_str.split(";"):
                 start, end = span.split()
                 ann["offsets"].append([int(start), int(end)])
 
+            # Heuristically split text of discontiguous entities into chunks
+            ann["text"] = []
+            if len(ann["offsets"]) > 1:
+                i = 0
+                for start, end in ann["offsets"]:
+                    chunk_len = end - start
+                    ann["text"].append(text[i:chunk_len+i])
```

Comment:
Isn't this similar to do?:
```python
texts = [doc_text[i:j] for i, j in offsets]
```
which if I understood correctly we do not want to do?

Reply:
@leonweber

Reply:
No, it shouldn't be. Because `text` is the entity text and not the document text. Or am I missing something?

Reply:
Pfff... sorry my bad. All fine for me then!